src/lib/extensions.ts

```diff
@@ -0,0 +1,27 @@
+import { OptionValues } from 'commander'
+import isEmpty from 'lodash/isEmpty.js'
+import BaseCommand from '../commands/base-command.js'
+import { init } from '../commands/init/init.js'
+import { log } from '../utils/command-helpers.js'
+
+export const packagesThatNeedSites = ['@netlify/neon']
+
+export async function handleExtensionRequirements(options: OptionValues, command: BaseCommand) {
+  const { project } = command
+  const { site, siteInfo } = command.netlify
+
+  const packageJson = await project.getPackageJSON()
+  const dependencies = packageJson.dependencies ?? {}
+
+  // If we don't have a site, these extensions need one initialized
+  const hasSiteData = Boolean(site.id ?? options.site) && !isEmpty(siteInfo)
+  if (!hasSiteData) {
+    for (const packageName of packagesThatNeedSites) {
+      if (dependencies[packageName]) {
+        log(`Found ${packageName} in package.json, initializing a site`)
+        await init(options, command)
```

Comment:
btw, passing the dev command's `options` as the `init` command's `options` only happens to work because we mostly use this very weak `BaseCommand` type instead of command-specific types.

I don't have a great suggestion here, honestly. It would be nice if we had a util to extract just the "global" shared options from an options bag, so you could pass those down but not the `dev`-specific ones.

🤷🏼 I guess just food for thought for later.